### PR TITLE
Add ML Model Search API

### PR DESCRIPTION
**Pull request by Zhangxunmt**
### Description
Add the search model API for ml-common.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/action/models/SearchModelTransportAction.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            log.error(e);
+            listener.onFailure(e);
+        }
+    }
+
+    /**
+     * Wrap action listener to avoid return verbose error message and wrong 500 error to user.
+     * Suggestion for exception handling in AD:
+     * 1. If the error is caused by wrong input, throw IllegalArgumentException exception.
+     * 2. For other errors, please use AnomalyDetectionException or its subclass, or use
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/action/models/SearchModelTransportAction.java:53]: Use MLException?

File: client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java [hunk 2]
```diff
@@ -10,15 +10,20 @@
...
 import org.opensearch.ml.common.parameter.Input;
 import org.opensearch.ml.common.parameter.MLInput;
 import org.opensearch.ml.common.parameter.MLOutput;
 import org.opensearch.ml.common.parameter.Output;
+import org.opensearch.ml.common.parameter.MLModel;
 import org.opensearch.ml.common.transport.MLTaskResponse;
 import org.opensearch.ml.common.transport.execute.MLExecuteTaskAction;
 import org.opensearch.ml.common.transport.execute.MLExecuteTaskRequest;
 import org.opensearch.ml.common.transport.execute.MLExecuteTaskResponse;
+import org.opensearch.ml.common.transport.model.*;
```
[the remark this answers is not in the document]
Comment (Zhangxunmt) [reply client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java:26]: this must be auto added in intellij.

File: plugin/src/main/java/org/opensearch/ml/action/models/SearchModelTransportAction.java (hunk 1 above)
Comment (Zhangxunmt) [reply plugin/src/main/java/org/opensearch/ml/action/models/SearchModelTransportAction.java]: Corrected!

**Review by ylwu-amzn — DISMISSED** [review]

File: common/src/test/java/org/opensearch/ml/common/transport/model/MLModelDeleteRequestTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
+package org.opensearch.ml.common.transport.model;
```
Comment (ylwu-amzn) [inline common/src/test/java/org/opensearch/ml/common/transport/model/MLModelDeleteRequestTest.java:1]: Add copyright header

Comment (Zhangxunmt) [reply common/src/test/java/org/opensearch/ml/common/transport/model/MLModelDeleteRequestTest.java:6]: added! Thanks for the remind.

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**